### ISSUE TITLE
Update to latest Nuget package throws exceptions for JSON POSTs

### ISSUE THREAD

**Issue by RyanONeill1970** [51Degrees/pipeline-dotnet#5]
I've been using the DeviceDetection and Pipeline.Web packages to detect screen width and the mobile flag. After updating to the latest version (see below) I now receive an error from FiftyOne when I post JSON to the website. It may be that I should move the endpoints that are triggering this to a class decorated with ApiController but this does not seem to exclude FiftyOne from trying to examine the form contents.

    <PackageReference Include="FiftyOne.DeviceDetection" Version="4.2.4" />
    <PackageReference Include="FiftyOne.Pipeline.Web" Version="4.2.3" />

It seems that this issue was introduced about 3 months ago when checking if the request type was a POST, but it did not check if the request was form-encoded and .Net falls over when the Form is enumerated on a non form-encoded POST.

The code that causes this is here;
https://github.com/51Degrees/pipeline-dotnet/blob/5f20c311f27a03c2ef4466f82f8a7b73ce8e67c6/Web%20Integration/FiftyOne.Pipeline.Web/Services/WebRequestEvidenceService.cs#L129-L137

It results in this exception;

```
fail: Microsoft.AspNetCore.Diagnostics.DeveloperExceptionPageMiddleware[1]
      An unhandled exception has occurred while executing the request.
      System.InvalidOperationException: Incorrect Content-Type: application/json; charset=UTF-8
         at Microsoft.AspNetCore.Http.Features.FormFeature.ReadForm()
         at Microsoft.AspNetCore.Http.DefaultHttpRequest.get_Form()
         at FiftyOne.Pipeline.Web.Services.WebRequestEvidenceService.AddEvidenceFromRequest(IFlowData flowData, HttpRequest httpRequest)
         at FiftyOne.Pipeline.Web.Services.PipelineResultService.Process(HttpContext context)
         at FiftyOne.Pipeline.Web.FiftyOneMiddleware.Invoke(HttpContext context)
         at Microsoft.AspNetCore.Diagnostics.DeveloperExceptionPageMiddleware.Invoke(HttpContext context)
```

**Comment by Piotr-51Degrees** [comment]
Hi @RyanONeill1970, 

Thank you for the issue report - we will have a look into this as soon as possible. I will keep you informed on our progress. 

Thanks,
Piotr

**Comment by Piotr-51Degrees** [comment]
Hi @RyanONeill1970 

Thanks again for reporting this issue - we have managed to identify the problem and we aim to deliver a fix for this in version 4.3.0 which is due to be released next week. I will let you know as soon as this is ready.

Thanks,
Piotr

**Comment by ben51degrees** [comment]
Closed with [release 4.3.0](https://github.com/51Degrees/pipeline-dotnet/releases/tag/4.3.0), with the specific commit info being 6f3c4977ce46654f102b5202d3c447b5f58be465.
